source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp

```diff
@@ -170,42 +169,44 @@ void bindEnvironmentLighting(MHWRender::MShaderInstance& shader,
     MHWRender::MTextureDescription textureDescription;
     if (parameterList.indexOf(mx::HW::ENV_IRRADIANCE.c_str()) >= 0)
     {
-        status = bindFileTexture(shader, mx::HW::ENV_IRRADIANCE, imageSearchPath, envIrradiancePath, samplerDescription, textureDescription, nullptr);
+        status = bindFileTexture(shader, mx::HW::ENV_IRRADIANCE, imageSearchPath,
+                                 node.getEnvIrradianceFileName().asChar(), samplerDescription, textureDescription, nullptr);
     }
 
     // Set radiance map
     if (parameterList.indexOf(mx::HW::ENV_RADIANCE.c_str()) >= 0)
     {
-        status = bindFileTexture(shader, mx::HW::ENV_RADIANCE, imageSearchPath, envRadiancePath, samplerDescription, textureDescription, nullptr);
+        status = bindFileTexture(shader, mx::HW::ENV_RADIANCE, imageSearchPath,
+                                 node.getEnvRadianceFileName().asChar(), samplerDescription, textureDescription, nullptr);
```

Comment:
If we change the attribute can we make this to trigger again. I think you did added in something to do this for but filter by the document + element attrs only. I assume we add in these new attrs then a rebuild will occur as well?

Reply:
Yes, refresh works for these as well thanks to `valueChangeRequiresFragmentRebuild()`